### ISSUE TITLE
[Linux] Graphical issues

### ISSUE THREAD

**Issue by BanhmiDev** [ValveSoftware/Dota-2#47]
Thumbnails are initially blurred out, although change to their original resolution when hovering over them
![Blurry](https://i.imgur.com/PmvHWYZ.png)

Heroes don't even show up in showcase mode
![Hero](https://i.imgur.com/NgEphEL.png)

edit: Games do not load - correlating to https://github.com/ValveSoftware/Dota-2-Reborn/issues/10

System specifications:

```
Processor Information:
    Vendor:  AuthenticAMD
    CPU Family:  0x10
    CPU Model:  0x2
    CPU Stepping:  0x3
    CPU Type:  0x0
    Speed:  2800 Mhz
    2 logical processors
    2 physical processors
    HyperThreading:  Unsupported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Unsupported
    SSE4a:  Supported
    SSE41:  Unsupported
    SSE42:  Unsupported

Operating System Version:
    Linux (64 bit)
    Kernel Name:  Linux
    Kernel Version:  4.0.5-1-ARCH
    X Server Vendor:  The X.Org Foundation
    X Server Release:  11702000
    X Window Manager:  Xfwm4
    Steam Runtime Version:  steam-runtime-release_2015-01-06

Video Card:
    Driver:  X.Org Gallium 0.4 on AMD BARTS

    Driver Version:  3.0 Mesa 10.6.0
    OpenGL Version: 3.0
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 60 Hz
    VendorID:  0x1002
    DeviceID:  0x6739
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    Primary Display Resolution:  1920 x 1080
    Desktop Resolution: 1920 x 1080
    Primary Display Size: 18.90" x 10.63"  (21.65" diag)
                                            48.0cm x 27.0cm  (55.0cm diag)
    Primary VRAM Not Detected
```

**Comment by fasmat** [comment]
I have the same issues. Heroes in the Showcase mode don't have any textures (but particles). Some cosmetic item icons are missing, others are blurry.

**Comment by gdrewb-valve** [comment]
Does the 6/24 update change anything?

**Comment by gimu** [comment]
Hero showcase now works fine, but the first issue still exists.
Almost every image is initially blurred out - the only way to resolve this was to switch between the dashboard scenes several times (which does not always work out).

Note: Even if the "All Heroes" thumbnails are getting back to their appropriate resolution, hovering over them doesn't resize their textures at all.
![Blurry](http://i.imgur.com/r63oK07.jpg)
![Blurry2](http://i.imgur.com/7GDWEOf.png)

**Comment by danginsburg** [comment]
I think the issue with the blurred heroes in the menu UI is probably due to you having Texture Quality set to "Low".  We bring in higher resolution versions when the Texture Quality is set to "High".  I did find one issue while investigating this which is that we were not detecting GPU memory size correctly on AMD mesa which could impact the size of textures allowed streamed in at one time, but I have fixed this using GLX_mesa_query_renderer.  I don't believe there is any Linux-specific issue remaining here.  The blurry heroes happens on Windows as well on low texture quality.
